:!sectids:

== Why is this an issue?

When using the Enum ToString() syntax to convert an enum value to a string, without any specific format, the result will be the same as using the nameof(value) syntax.
While the first one is resolved at runtime and requires some work, the latter is resolved at compile time and is much more efficient.

== How to fix it
=== Noncompliant code example

[source, cs]
----
private enum Letter { A, B, C }

public void Main()
{
    Console.WriteLine(Letter.A.ToString()); // Noncompliant, use nameof
}
----

=== Compliant solution

[source, cs]
----
private enum Letter { A, B, C }

public void Main()
{
    Console.WriteLine(Letter.A.ToString("D")); // Compliant, the string format impacts the result
}
----

[source, cs]
----
public void Main()
{
    Console.WriteLine(nameof(Letter.A)); // Compliant
}
----

== Resources

Reference/Validation is unknown.
