:!sectids:

== Why is this an issue?

SVG images generated by common drawing softwares contains unnecessary data: calc layer, metadata, namespaces and comments.

== How to fix it
=== Noncompliant code example

[source,xml]
----
<!-- Noncompliant -->
<!-- Created with Inkscape (http://www.inkscape.org/) -->
<svg
    width="210mm"
    height="297mm"
    viewBox="0 0 210 297"
    version="1.1"
    id="svg5"
    inkscape:version="1.2.2 (b0a8486541, 2022-12-01)"
    sodipodi:docname="dessin2.svg"
    xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
    xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:svg="http://www.w3.org/2000/svg">
    <g
            inkscape:label="Calque 1"
            inkscape:groupmode="layer"
            id="layer1">
        <circle
            style="fill:#ff00ff;stroke-width:0.264583"
            id="path111"
            cx="104.02724"
            cy="152.19028"
            r="73.177132" />
    </g>
</svg>
----

=== Compliant solution

[source,xml]
----
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 210 297" width="210mm" height="297mm">
    <circle cx="104.02724" cy="152.19028" r="73.177132" style="fill:#ff00ff;stroke-width:0.264583"/>
</svg>
----

== Resources

- https://github.com/cnumr/best-practices/blob/main/chapters/BP_036_fr.md[CNumR best practices (3rd edition) BP_036]
