:!sectids:

== Why is this an issue?

Turning on the torch mode programmatically with `react-native-torch` must absolutely be avoided because the flashlight is one of the most energy-intensive component.

== How to fix it

=== Noncompliant code example

[source,js,data-diff-id="2",data-diff-type="noncompliant"]
----
import Torch from 'react-native-torch'; // Noncompliant
----

=== Compliant solution

Do not set the torch programmatically.

== Resources

=== Documentation

- [CNumR best practices mobile](https://github.com/cnumr/best-practices-mobile) - Torch free