:!sectids:

== Why is this an issue?


Do not call a function when declaring a for-type loop in order to avoid function calls each iteration. It saves CPU cycles.

== How to fix it
=== Noncompliant code example

[source,php]
----
for ($i = 0; $i <= foo(); $i++) {  // Noncompliant
	// ......
}
----

=== Compliant solution

[source,php]
----
$maxI = foo();
for ($i = 0; $i <= $maxI; $i++) {
  .....
}

  OR

for ($i = 0, $maxI = foo(); $i <= $maxI; $i++) {
  .....
}
----

== Resources

=== Best practises

- https://www.greenit.fr/2019/05/07/ecoconception-web-les-115-bonnes-pratiques-3eme-edition/[CNumR best practices (3rd edition) BP_069 (no longer exists in edition 4)]

=== Impact

include::../../etsdiff-methodology.asciidoc[]

==== Case for a 1GB database

image::https://live.staticflickr.com/65535/52622634654_bf3c3d9ba8_o.png[ETSdiff percent comparison]

[format=csv,cols="1h,1,1"]
|===
Source of impacts,Compliant,Noncompliant

include::1GB.etsdiff.csv[]
|===
