:!sectids:

== Why is this an issue?

Most Android-powered devices have built-in sensors that measure motion, orientation, and various environmental conditions.
    In addition to these are the image sensor (a.k.a. Camera) and the geo-positioning sensor (a.k.a. GPS).

The common point of all these sensors is that they are expensive while in use. Their common bug is to let the sensor unnecessarily process data when the app enters an idle state, typically when paused or stopped.

    Consequently, calls must be carefully pairwised: `LocationManager.requestLocationUpdates()/removeUpdates()`.

    Failing to do so can drain the battery in just a few hours.

== How to fix it
=== Noncompliant code example

[source,java]
----
// Noncompliant
LocationManager locationManager = (LocationManager) getSystemService(LOCATION_SERVICE);
locationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER, 1000L, 1.0f, this);
----

=== Compliant solution

[source,java]
----
LocationManager locationManager = (LocationManager) getSystemService(LOCATION_SERVICE);
locationManager.requestLocationUpdates(LocationManager.GPS_PROVIDER, 1000L, 1.0f, this);
locationManager.removeUpdates(this);
----

== Resources

- https://github.com/cnumr/best-practices-mobile[Mobile-specific Best Practices] by https://collectif.greenit.fr/index_en.html[CNumR]
