<p>If possible, the utilisation of svg image format (or &lt;svg/&gt; html tag) is recommended over other image format.</p>
<p>Because SVGs are generally smaller than other image format, theyâ€™re less taxing on your server despite needing to render on load.</p>
<p>When to use SVG :
<ul>
    <li>Your image is used for decorative website graphics, logos, icons, graphs and diagrams, and other simple images.</li>
    <li>You image require animation.</li>
    <li>You image need to be responsive and scale without lack of quality.</li>
</ul>
</p>
<p>Some advantages of using SVG :
<ul>
    <li>SVGs are scalable and will render pixel-perfect at any resolution whereas JPEGs, PNGs and GIFs will not.</li>
    <li>SVGs are vector images and therefore are usually much smaller in file-size than bitmap-based images.</li>
    <li>SVGs can be embedded into the HTML which means they can be cached, edited directly using CSS and indexed for greater accessibility.</li>
    <li>SVGs can be animated directly or by using CSS or JavaScript making it easy for web designers to add interactivity to a site.</li>
</ul>
</p>

<h2>Noncompliant Code Example</h2>
<pre>
    ...
    img_jpg = "image.jpg"
    ...
</pre>
<h2>Compliant Solution</h2>
<pre>
    ...
    img_svg = "image.svg"
    ...
</pre>
<h2>Noncompliant Code Example</h2>
<pre>
    public void foo() {
        ...
        image_format = testImage("image.jpg")
        ...
    }
</pre>
<h2>Compliant Solution</h2>
<pre>
    public void foo() {
        ...
        image_format = testImage("image.svg")
        ...
   }
</pre>
