<p>Databases servers have to solve fields regarding the schema. Knowing and using the schema save CPU cycles and network transfer.</p>
<h2>Noncompliant Code Example</h2>
<pre>
    public void foo() {
        ...
        String baseQuery = "SELECT * FROM users"; // Noncompliant

        ...
    }

</pre>
<h2>Compliant Solution</h2>
<pre>

    public void foo() {
        ...
        String query = "SELECT id,name, adress FROM users ";
        ...
   }

</pre>
