:!sectids:

== Why is this an issue?

Prefer local variables as parameters

When calling a global variable, the interpretation engine must check that it exists in all the scopes, that it has a value, etc.
Passing global variables as arguments gives them the status of local variables inside the function, thus saving computing time (CPU cycles).

== How to fix it

=== Noncompliant code example

[source,php]
----
// Noncompliant
var aGlobal = new String('Hello');

function globalLength(){
    length = aGlobal.length;
    console.log(length);
}

globalLength();
----

=== Compliant solution

[source,php]
----
var aGlobal = new String('Hello');

function someVarLength(str){
    length = str.length;
    console.log(length);
}

somVarLength(aGlobal);
----

== Resources

- https://www.greenit.fr/2019/05/07/ecoconception-web-les-115-bonnes-pratiques-3eme-edition/[CNumR best practices (3rd edition) BP_050 (no longer exists in edition 4)]
