:!sectids:

== Why is this an issue?

To avoid draining the battery, an Android device that is left idle quickly falls asleep.

Hence, keeping the screen on should be avoided, unless it is absolutely necessary.

To keep the screen on, developers typically:

- use the `FLAG_KEEP_SCREEN_ON` in their activity;
- or use the `android:keepScreenOn` attribute in their application's layout XML file.

== How to fix it
=== Noncompliant code example

[source,java]
----
// Noncompliant
getWindow().setFlags(FLAG_KEEP_SCREEN_ON, FLAG_KEEP_SCREEN_ON);
----

=== Compliant solution

Do not keep the screen on.

== Resources

- https://github.com/cnumr/best-practices-mobile[Mobile-specific Best Practices] by https://collectif.greenit.fr/index_en.html[CNumR]
