:!sectids:

== Why is this an issue?

To avoid draining the battery, an iOS device that is left idle quickly falls asleep.

Hence, keeping the screen on should be avoided, unless it is absolutely necessary.
Developers typically disable the idle timer with `UIApplication.shared.isIdleTimerDisabled = true`.

== How to fix it

=== Noncompliant code example

[source,swift]
----
// Noncompliant
UIApplication.shared.isIdleTimerDisabled = true
----

=== Compliant solution

Do not disable idle timer.

== Resources

- https://github.com/cnumr/best-practices-mobile[Mobile-specific Best Practices] by https://collectif.greenit.fr/index_en.html[CNumR]
