:!sectids:

== Why is this an issue?

Taking into account when the device is entering or exiting the power save mode is higly desirable for the battery life.

It implies the existence of a broadcast receiver registered on the action `ACTION_POWER_SAVE_MODE_CHANGED`, or programmatically with a call to `PowerManager#isPowerSaveMode().`

== How to fix it
=== Noncompliant code example

[source,xml]
----
// Noncompliant
// No BATTERY_CHANGED intent
----

=== Compliant solution

[source,xml]
----
<receiver android:name="io.ecocode.testapp.core.service.MyReceiver">
    <intent-filter>
        <action android:name="android.intent.action.BATTERY_CHANGED"/>
    </intent-filter>
</receiver>
----

== Resources

- https://github.com/cnumr/best-practices-mobile[Mobile-specific Best Practices] by https://collectif.greenit.fr/index_en.html[CNumR]
