:!sectids:

== Why is this an issue?

Comparing a string to an empty string is unnecessary and can be replaced by a call to `string.Length`, which is more performant and more readable.

=== When can it be ignored?

This rule should not be ignored.

== How to fix it
=== Noncompliant code example

[source, cs]
----
public void Main()
{
	string s = "";
	if (s == "") // Noncompliant
	{
		Console.WriteLine("Empty string");
	}
}
----

=== Compliant solution

[source, cs]
----
 public void Main()
{
	string s = "";
	if (s.Length == 0) // Compliant
	{
		Console.WriteLine("Empty string");
	}
}
----

== Resources

Reference/Validation is unknown.
