:!sectids:

== Why is this an issue?

To avoid draining the battery, an Android device that is left idle quickly falls asleep.

Hence, keeping the screen on should be avoided, unless it is absolutely necessary.

Bad practise includes using a Power Manager system service feature called wake locks
by invoking `PowerManager.WakeLock#newWakeLock(int levelAndFlags, String tag)`,
along with the specific permission `WAKE_LOCK` in the manifest.

== How to fix it
=== Noncompliant code example

[source,xml]
----
<!-- Noncompliant -->
<manifest>
    <uses-permission android:name="android.permission.WAKE_LOCK" />
</manifest>
----

=== Compliant solution

Do not use wake lock.

== Resources

- https://github.com/cnumr/best-practices-mobile[Mobile-specific Best Practices] by https://collectif.greenit.fr/index_en.html[CNumR]
