<p>The form $i++ creates a temporary variable whereas ++$i does not. It save CPU cycles.</p>
<h2>Noncompliant Code Example</h2>
<pre>
i++  // Noncompliant
</pre>
<h2>Compliant Solution</h2>
<pre>++i</pre>
