:!sectids:

== Why is this an issue?


Executing SQL queries in loop induced unnecessary calculation by the CPU, RAM usage and network transfer.

== How to fix it
=== Noncompliant code example

[source,python]
----
def foo():
    ...
    results = []
    for id in range(20):
      results.append(cursor.execute("SELECT name FROM users where id = ?", (id)).fetchone()) # Noncompliant {{Avoid performing SQL queries within a loop}}
    ...
----

=== Compliant solution

[source,python]
----
def foo():
    ...
    ids = range(20)
    results = cursor.execute("SELECT name FROM users where id IN ({0})".format(', '.join("?" * len(ids))), ids).fetchmany() # Compliant
    ...
----

== Resources

- https://github.com/cnumr/best-practices/blob/main/chapters/BP_072_fr.md[CNumR best practices (3rd edition) BP_072]
