:!sectids:

== Why is this an issue?

Avoid using getters and setters in a class, as they increase unnecessary RAM memory usage.

== How to fix it
=== Noncompliant code example

[source,python]
----
# Noncompliant
class Client():

    def __init__(self, age):
        self.age = age

    def get_age(self):
        return self.age

    def set_age(self, age):
        self.age = age

client = Client(25)
client.get_age() # Useless getter
client.set_age(26) # Useless setter
----

=== Compliant solution

[source,python]
----
class Client():

    def __init__(self, age):
        self.age = age

client = Client(25)
client.age
client.age = 26
----

== Resources

- https://www.greenit.fr/2019/05/07/ecoconception-web-les-115-bonnes-pratiques-3eme-edition/[CNumR best practices (3rd edition) BP_062 (no longer exists in edition 4)]
