:!sectids:

== Why is this an issue?

Throwing and handling exceptions is expensive and should not be used for normal program flows.

== How to fix it
=== Noncompliant code example

[source,java]
----
// Noncompliant
public void readPreferences(String filename) {
    //...
    InputStream in = null;
    try {
        in = new FileInputStream(filename);
    } catch (FileNotFoundException e) {
        logger.log(e);
    }
    in.read(...);
    //...
}
----

=== Compliant solution

[source,java]
----
public void readPreferences(String filename) throws IllegalArgumentException, FileNotFoundException, IOException {
    if (filename == null) {
        throw new IllegalArgumentException ("filename is null");
    }
    //...
    InputStream in = new FileInputStream(filename);
    //...
}
----

== Resources

Reference/Validation for this rule is unknown.
