:!sectids:

== Why is this an issue?

Use `try-with-resources` statement for any object that implements the `AutoCloseable` interface, it save computer resources.

== How to fix it
=== Noncompliant code example

[source,java]
----
// Noncompliant
private static void printFileJava7() throws IOException {
    FileInputStream input = new FileInputStream("file.txt");
    int data = input.read();
    while(data != -1){
        System.out.print((char) data);
        data = input.read();
    }
}
----

=== Compliant solution

[source,java]
----
private static void printFileJava7() throws IOException {
    try(FileInputStream input = new FileInputStream("file.txt")) {
        int data = input.read();
        while(data != -1){
            System.out.print((char) data);
            data = input.read();
        }
    }
}
----

== Resources

Reference/Validation is unknown.
