:!sectids:

== Why is this an issue?

Database servers have to resolve schema fields when using asterisk symbol (`*`). Knowing and using the schema saves CPU cycles and network transfer.

== How to fix it
=== Noncompliant code example

[source,python]
----
public void foo() {
    ...
    String baseQuery = "SELECT * FROM users"; // Noncompliant
    ...
}
----

=== Compliant solution

[source,python]
----
public void foo() {
    ...
        String query = "SELECT id, name, address FROM users ";
    ...
}
----

== Resources

- https://www.greenit.fr/2019/05/07/ecoconception-web-les-115-bonnes-pratiques-3eme-edition/[CNumR best practices (3rd edition) BP_074 (no longer exists in edition 4)]
